tasks/pkg-redhat.yml

```diff
@@ -41,7 +41,21 @@
 
 # Using `yum` directly to work around an Ansible bug:
 # https://github.com/ansible/ansible/issues/20608
-- name: Install latest datadog-agent package
+- name: Install latest datadog-agent package issue 20608 workaround
   command: yum install -y datadog-agent
-  when: datadog_agent_version == ""
+  register: yum_install_result
+  changed_when: ( yum_install_result.stdout_lines | last ) != "Nothing to do"
+  when:
+  - datadog_agent_version == ""
+  - ansible_version is version_compare(2.3, '<')
```

Comment:
We actually only support >= 2.4: https://github.com/DataDog/ansible-datadog/blob/ec5e5fd736982e87d4a31f06a8ba484f97ab8340/meta/main.yml#L6 so we can remove that case and keep the one below given that the `#20608` issue is now resolved in >= 2.4.

Reply:
Good idea, much simpler